### PR TITLE
Add polish language support

### PR DESCRIPTION
**Pull request by ertrzyiks**
Add polish language support #292 

It will be more complicated than that because of our declination. 

Update:
Ready for review :)

### CODE REVIEW

File: src/locale/pl/build_format_locale/index.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+var buildFormattingTokensRegExp = require('../../_lib/build_formatting_tokens_reg_exp/index.js')
+
+function buildFormatLocale () {
+  // Note: in English, the names of days of the week and months are capitalized.
+  // If you are making a new locale based on this one, check if the same is true for the langugage you're working on.
+  // Generally, formatted dates should look like they are in the middle of a sentence.
+  // e.g. in Spanish language the weekdays and months should be in the lowercase.
+  var months3char = ['Sty', 'Lut', 'Mar', 'Kwi', 'Maj', 'Cze', 'Lip', 'Sie', 'Wrz', 'Paź', 'Lis', 'Gru']
```
Comment (LeshaKoss) [inline src/locale/pl/build_format_locale/index.js]: Please capitalize the month and the weekdays names only if they are generally capitalized in the middle of sentence (like in English and German)

Comment (piotrd) [inline src/locale/pl/build_format_locale/index.js]: True, in polish these would be written in lowercase.

**Comment by LeshaKoss** [comment]
@ertrzyiks thank you for your contribution!

> It will be more complicated than that because of our declination.

You may want to look how we solved declination problem in the russian locale

For example, formats "D MMMM", "Do MMMM" and "DD MMMM" like "6 December" force genitive case on the month name, so we implemented separate "D MMMM", "Do MMMM" and "DD MMMM" formatters:
https://github.com/date-fns/date-fns/blob/master/src/locale/ru/build_format_locale/index.js#L77

In distanceInWords, we had to have up to nine formats for a single token: https://github.com/date-fns/date-fns/blob/master/src/locale/ru/build_distance_in_words_locale/index.js#L52 and choose it with `declension` function: https://github.com/date-fns/date-fns/blob/master/src/locale/ru/build_distance_in_words_locale/index.js#L1

**Comment by ertrzyiks** [comment]
@LeshaKoss thanks for tips, I will follow the same approach for future and past cases in polish language

**Comment by ertrzyiks** [comment]
@LeshaKoss current build should be green. However. it deserves a refactor and I still don't know what to do with 
```
var meridiemUppercase = ['AM', 'PM']
var meridiemLowercase = ['am', 'pm']
var meridiemFull = ['a.m.', 'p.m.']
```
We dont use those in polish languge. I saw that you asked for verbose version for french locale, so the same here? just 'morning' and 'afternoon' for meridiemFull?

**Comment by LeshaKoss** [comment]
@ertrzyiks yeah, maybe even "night"/"morning"/"day"/"afternoon" if it applies in Polish language (like here: https://github.com/date-fns/date-fns/blob/master/src/locale/ru/build_format_locale/index.js#L40)

**Comment by ertrzyiks** [comment]
@LeshaKoss @kossnocorp I think it's ready. Yes, `"night"/"morning"/"day"/"evening"` applies in polish language and it has been addressed already [here](https://github.com/date-fns/date-fns/pull/294/files#diff-de10507e3a0f993eadf1a603fd794446R9)

Sorry for so huge diff, I warned you it might be complicated in the end :D I will review myself again tomorrow with a fresh mind. @piotrd maybe you want to help me with the review?

File: src/locale/pl/build_distance_in_words_locale/index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      case 3:
+      case 4:
+        return scheme.two_four
+    }
+  }
+
+  return scheme.other
+}
+
+function declension (scheme, count, time) {
```
Comment (LeshaKoss) [inline src/locale/pl/build_distance_in_words_locale/index.js]: @ertrzyiks monumental work! 👍

File: src/locale/pl/build_distance_in_words_locale/index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    return scheme.one
+  }
+
+  var rem100 = count % 100
+  if (rem100 > 20 || rem100 < 10) {
+    switch (rem100 % 10) {
+      case 2:
+      case 3:
+      case 4:
+        return scheme.two_four
```
Comment (LeshaKoss) [inline src/locale/pl/build_distance_in_words_locale/index.js]: Please convert `two_four` to camelCase 🙏

**Comment by ertrzyiks** [comment]
Self-review finished and translation seems to be ok. I've applied requested changes + added small improvement (get rid of switch-case).

**Comment by piotrd** [comment]
@ertrzyiks apologies, I didn't see the update earlier. I can review it tomorrow, as today it's already a bit late.

File: src/locale/pl/test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// @flow
+/* eslint-env mocha */
+
+var assert = require('power-assert')
+var plLocale = require('./')
+
+describe('en locale', function () {
```
Comment (piotrd) [inline src/locale/pl/test.js]: Shouldn't this say `'pl locale'` ?

**Comment by LeshaKoss** [comment]
@ertrzyiks thanks again for your awesome contribution! 🇵🇱 Polish locale was released with [1.17.0](https://github.com/date-fns/date-fns/releases/tag/v1.17.0) (diff: https://github.com/date-fns/date-fns/compare/v1.16.0...v1.17.0)

**Pull request merged**